src/components/Messaging/Survey.js

```diff
@@ -158,8 +167,41 @@ const SurveyForm = ({ setToggleSurvey }) => {
         payload.survey.title.length > 1 &&
         (payload['region_id'] || payload['organization_id'])
       ) {
-        await postBulkMessageSend(payload);
-        history.go(0);
+        const res = await postBulkMessageSend(payload);
+        log.debug('Survey submit', threads, res);
+
+        if (res.error) {
+          setErrorMessage(res.message);
+        } else {
+          const newSurvey = {
+            id: uuid(),
+            type: 'survey',
+            parent_message_id: null,
+            from: payload.author_handle,
+            to: null,
+            recipient_organization_id: payload.organization_id || null,
+            recipient_region_id: payload.region_id || null,
+            subject: payload.subject,
+            body: null,
+            composed_at: new Date().toISOString(),
+            video_link: null,
+            survey_response: null,
+            survey: payload.survey,
+          };
+          log.debug('...update threads w/ new survey');
+          // update the full set of threads
+          setThreads((prev) => {
+            const updated = [
+              {
+                username: `${newSurvey.id}`,
+                messages: [newSurvey],
+                avatar: '',
+              },
+              ...prev,
+            ];
+            return updated;
```

Comment:
You can return directly the object instead of saving it in a variable and returning the variable